src/bindgen/parser.rs

```diff
@@ -636,36 +642,80 @@ impl Parse {
         }
     }
 
-    /// Loads an associated `const` declaration
-    fn load_syn_assoc_const(
+    /// Loads associated `const` declarations
+    fn load_syn_assoc_consts<'a, I>(
         &mut self,
         binding_crate_name: &str,
         crate_name: &str,
         mod_cfg: &Option<Cfg>,
         impl_ty: &syn::Type,
-        item: &syn::ImplItemConst,
-    ) {
-        if crate_name != binding_crate_name {
-            info!(
-                "Skip {}::{} - (const's outside of the binding crate are not used).",
-                crate_name, &item.ident
-            );
+        items: I,
+    ) where
+        I: IntoIterator<Item = &'a syn::ImplItemConst>,
+    {
+        let ty = Type::load(&impl_ty).unwrap();
+        if ty.is_none() {
             return;
         }
 
-        let const_name = item.ident.to_string();
+        let impl_struct_path = ty.unwrap().get_root_path().unwrap();
 
-        match Constant::load_assoc(const_name.clone(), item, impl_ty, mod_cfg) {
-            Ok(constant) => {
-                info!("Take {}::{}.", crate_name, &item.ident);
+        let is_transparent = if let Some(ref impl_items) = self.structs.get_items(&impl_struct_path)
+        {
+            if impl_items.len() != 1 {
+                error!(
+                    "Expected one struct to match path {}, but found {}",
+                    impl_struct_path,
+                    impl_items.len()
+                );
+                return;
+            }
+            if let ItemContainer::Struct(ref s) = impl_items[0] {
+                s.is_transparent
+            } else {
+                info!(
+                        "Skip impl block for {}::{} (impl blocks for associated constants are only defined for structs).",
+                        crate_name, impl_struct_path
+                    );
+                return;
+            }
+        } else {
+            error!(
+                    "Cannot find type for {}::{} (impl blocks require the struct declaration to be known).",
```

Comment:
Was it possible for users to have associated constants for `enum`s and not just `struct`s? It looks like the code didn't check for this constraint before, but I may be missing it.

If so, then this is kind of a breaking change. Not sure if that matters or not.

Reply:
I tried adding it to traits but it doesn't seem like there is much  in place for handling traits yet, (unless I've missed it). For example, I couldn't see how to know whether a trait has been implemented for a particular type, and without that I can't emit the associated constant for the concrete type.